aws/resource_aws_instance.go

```diff
@@ -1528,6 +1543,46 @@ func readSecurityGroups(d *schema.ResourceData, instance *ec2.Instance, conn *ec
 	return nil
 }
 
+func readPasswordData(d *schema.ResourceData, instance *ec2.Instance, conn *ec2.EC2) error {
```

Comment:
We should try to reduce this function's signature to one action and leave the state management outside of this function, e.g. 

```go
func getAwsEc2InstancePasswordData(instanceID string, conn *ec2.EC2) (string, error)
```